xla/stream_executor/rocm/hip_blas_lt.cc

```diff
@@ -390,6 +390,17 @@ absl::Status BlasLt::MatmulPlan::DoMatmul(
     DeviceMemoryBase b_scale, DeviceMemoryBase c_scale,
     DeviceMemoryBase d_scale, DeviceMemoryBase d_amax,
     blas::ProfileResult* profile_result) const {
+  {
+    absl::MutexLock lock{&blas_lt_ref_.parent_->LoggerMutex()};
+    using sei = internal::StreamExecutorInterface;
+
+    if(blas_lt_ref_.parent_->GetArgumentLoggingMode() & sei::ArgumentLogging::kGemm) {
+      blas_lt_ref_.parent_->ArgumentLogs().push_back(
+        /* Known bug: blas::CallContext is not passed to this method */
+          sei::GemmCallTrace{sei::GemmCallTrace::GemmType::kBlasLt, 0,
+            a.size(), b.size()});
+    }
```

Comment:
This should be replaced by a proper logging API on Stream executor rather than exposing details of the logging to the client.  What if we decided to replace the logging with a lockless implementation?  This also is hard to do static analysis on everywhere to prove the locking protocols are always correct.